jenkins/check_projects.py

```diff
@@ -39,190 +39,414 @@
-      updated.add(project)
-      return
-
-  with lock:
-    print >>sys.stderr, 'could not update ',
-    print project
-    errors.add(project)
-
-
-def Check(project, needed, mutate):
-  try:
-    with semaphore:
-      out = subprocess.check_output([
-          'gcloud',
-          'projects',
-          'get-iam-policy',
-          project,
-          '--format=json(bindings)',
-      ])
-  except subprocess.CalledProcessError:
-    with lock:
-      print >>sys.stderr, 'cannot access ',
-      print project
-      errors.add(project)
-      return
-
-  bindings = json.loads(out)
-  fixes = {}
-  roles = set()
-  for binding in bindings['bindings']:
-    role = binding['role']
-    roles.add(role)
-    members = binding['members']
-    if role in needed:
-      missing = set(needed[role]) - set(members)
-      if missing:
-        fixes[role] = missing
-    if role == 'roles/owner':
-      with lock:
-        helpers[project] = members
-  missing_roles = set(needed) - roles
-  for role in missing_roles:
-    fixes[role] = needed[role]
-
-  if not fixes:
-    with lock:
-      print >>sys.stderr, project, 'already configured'
-      completed.add(project)
-      return
-
-  if not mutate:
-    with lock:
-      print >>sys.stderr, 'Will not --fix ',
-      print project
-      print >>sys.stderr, '  wanted fixes: ', fixes
-      errors.add(project)
-      return
-
-  updates = []
-  for role, members in sorted(fixes.items()):
-    updates.extend(
-        threading.Thread(target=Update, args=(project, role, m))
-        for m in members
-    )
-
-  for update in updates:
-    update.start()
-
-  for update in updates:
-    update.join()
+            project
+        ]
+        err = self.rl_exec.call(cmdline, stdout=open('/dev/null', 'w'))
+        if not err:
+            _log.info('Added %s as %s to %s', member, role, project)
+            self.results.report_updated(project)
+        else:
+            _log.info('Could not update IAM for %s', project)
+            self.results.report_error(
+                project, 'update %s (role=%s, member=%s)' %
+                (self.name(), role, member))
+
+
+class EnableVmZonalDNS(ProjectProperty):
+    """Project has Zonal DNS enabled."""
+
+    def __init__(self, rl_exec, results):
+        self.rl_exec = rl_exec
+        self.results = results
+
+    def name(self):
+        return 'EnableVMZonalDNS'
+
+    def check_and_maybe_update(self, config, project, fix):
+        try:
+            out = self.rl_exec.check_output([
+                'gcloud', 'compute', 'project-info', 'describe',
+                '--project=' + project,
+                '--format=json(commonInstanceMetadata.items)'])
+        except subprocess.CalledProcessError:
+            _log.info('Cannot access %s', project)
+            return
+
+        enabled = False
+        metadata = json.loads(out)
+        if metadata and metadata['commonInstanceMetadata'] \
```

Comment:
Please make use of python's implicit line joining inside of parens rather than `\`:
```
if (metadata and m['cIM']
    and m['cIM']['items']):
  pass
```

https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length